edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleClient.cs

```diff
@@ -48,9 +55,41 @@ public class ModuleClient : IModuleClient
 
         public void Dispose() => this.deviceClient.Dispose();
 
-        public Task<Twin> GetTwinAsync() => this.deviceClient.GetTwinAsync();
+        public async Task<Twin> GetTwinAsync()
```

Comment:
Can the `SetMethodHandlerAsync` and `SetDesiredPropertyUpdateCallbackAsync` also throw object disposed?

Reply:
ModuleClient.Create will also shut down EdgeAgent if there is any exception. So these 2 should be covered as well.